src/components/CommentForm/CommentForm.js

```diff
@@ -1,76 +1,84 @@
-import React, { useState } from 'react';
-import bemCssModules from 'bem-css-modules';
-import { default as CommentFormStyles } from './CommentForm.module.scss';
-import { useDispatch } from 'react-redux';
-import { v4 as uuidv4 } from 'uuid';
-
-import { addComment } from '../../redux/actions';
+import React, { useState } from "react";
+import bemCssModules from "bem-css-modules";
+import { default as CommentFormStyles } from "./CommentForm.module.scss";
+import { useDispatch, useSelector } from "react-redux";
+import { addComment } from "../../redux/actions";
+import { v4 as uuidv4 } from "uuid";
+import { selectors } from "../../redux/selectors";
 
 const style = bemCssModules(CommentFormStyles);
 
-const CommentForm = ({
-    postId,
-    currentComments,
-    setCurrentComments,
-    setIsVisible }) => {
-    const dispatch = useDispatch();
-
-    const [formEmail, setFormEmail] = useState('');
-    const [formName, setFormName] = useState('');
-    const [formBody, setFormBody] = useState('');
-
-    const handleOnChangeEmail = e => setFormEmail(e.target.value);
-    const handleOnChangeName = e => setFormName(e.target.value);
-    const handleOnChangeBody = e => setFormBody(e.target.value);
+const CommentForm = ({ postId, setIsVisible }) => {
+  const isAddCommentError = useSelector(selectors.getCommentsError);
 
-    const handleAddCommentSubmit = postId => e => {
-        e.preventDefault();
+  const dispatch = useDispatch();
 
-        const newComment = {
-            postId,
-            id: uuidv4(),
-            name: formName,
-            email: formEmail,
-            body: formBody,
-        };
+  const [formEmail, setFormEmail] = useState("");
+  const [formName, setFormName] = useState("");
+  const [formBody, setFormBody] = useState("");
 
-        dispatch(addComment(newComment));
+  const handleOnChangeEmail = (e) => setFormEmail(e.target.value);
+  const handleOnChangeName = (e) => setFormName(e.target.value);
+  const handleOnChangeBody = (e) => setFormBody(e.target.value);
 
-        let newCurrentComments = [...currentComments];
-        newCurrentComments.push(newComment);
-        setCurrentComments(newCurrentComments);
+  const handleAddCommentSubmit = (postId) => (e) => {
+    e.preventDefault();
 
-        setFormEmail("");
-        setFormName("");
-        setFormBody("");
+    const id = uuidv4();
 
-        setIsVisible(false);
-    };
+    dispatch(addComment(postId, id, formName, formEmail, formBody));
 
+    if (!isAddCommentError) {
+      setFormEmail("");
+      setFormName("");
+      setFormBody("");
```

Comment:
Reset formularza wywoła się niezależnie od tego, czy w akcji `addComment` otrzymamy błąd z API, czy nie. Warunek `!isAddCommentError` nie zadziała poprawnie, ponieważ ten kod wykonuje się synchronicznie, podczas gdy request do API w `addComment` jest asynchroniczny - to znaczy, że ustawienie błędu: `.catch(error => { dispatch(commentAddedFail(error)) })` wykona się **później**, niż sprawdzenie warunku `if (!isAddCommentError)`. 

Rozwiązaniem może być
- zamknięcie instrukcji resetowania w funkcji, przekazanie jej w argumencie skacji `addComment` i wywołanie jako callback w bloku `then`,
- wykorzystanie stanu `getAllComments`, aby wykryć nowo dodany komentarz w hooku `useEffect` i tam zresetować formularz.